darts/timeseries.py

```diff
@@ -5659,6 +5659,39 @@ def concatenate(
     return TimeSeries.from_xarray(da_concat, fill_missing_dates=False)
 
 
+def intersect(series: Sequence[TimeSeries]):
+    """Returns the intersection with respect to the time index of multiple ``TimeSeries``.
+
+    Parameters
+    ----------
+    series : Sequence[TimeSeries]
+        sequence of ``TimeSeries`` to intersect
+
+    Returns
+    -------
+    Sequence[TimeSeries]
+        Intersected series
+    """
+    if not series:
+        return []
+
+    data_arrays = []
+    has_datetime_index = series[0].has_datetime_index
+    for ts in series:
+        if ts.has_datetime_index != has_datetime_index:
+            raise_log(
+                IndexError(
```

Comment:
We almost always use `ValueError` in Darts, I think we should keep it that way for now
 
```suggestion
                ValueError(
```